diff -urN sqlite-3.5.1.org/www/docs.tcl sqlite-3.5.1/www/docs.tcl
--- sqlite-3.5.1.org/www/docs.tcl	2007-10-03 23:20:33.000000000 +0200
+++ sqlite-3.5.1/www/docs.tcl	2007-10-20 23:00:41.453214050 +0200
@@ -50,12 +50,12 @@
   A description of the TCL interface bindings for SQLite.
 }
 
-doc {How SQLite Implements Atomic Commit} {
+doc {How SQLite Implements Atomic Commit} {ac/atomiccommit.html} {
   A description of the logic within SQLite that implements
   transactions with atomic commit, even in the face of power
   failures.
 }
-doc {Moving From SQLite 3.4 to 3.5} {
+doc {Moving From SQLite 3.4 to 3.5} {34to35.html} {
   A document describing the differences between SQLite version 3.4.2
   and 3.5.0.
 }
