diff -ur sqlite-3.2.7-orig/test/conflict.test sqlite-3.2.7/test/conflict.test
--- sqlite-3.2.7-orig/test/conflict.test        2005-09-20 18:46:02.000000000 +0200
+++ sqlite-3.2.7/test/conflict.test     2005-09-26 12:45:00.245742000 +0200
@@ -302,6 +302,7 @@
     db close
     sqlite3 db test.db
     if {$conf1!=""} {set conf1 "ON CONFLICT $conf1"}
+    execsql {pragma temp_store=file}
     set ::sqlite_opentemp_count 0
     set r0 [catch {execsql [subst {
       DROP TABLE t1;

